modules/fitsfilereader/rendering/renderabletimevaryingfitssphere.cpp

```diff
@@ -0,0 +1,710 @@
+            "Source folder for RenderableTimeVaryingFitsSphere contains no files"
+        );
+    }
+    else {
+        computeSequenceEndTime();
+    }
+}
+
+void RenderableTimeVaryingFitsSphere::update(const UpdateData& data) {
+    RenderableSphere::update(data);
+
+    const double currentTime = data.time.j2000Seconds();
+    const double deltaTime = global::timeManager->deltaTime();
+
+    if (!_dynamicFileDownloader && _loadingType == LoadingType::DynamicDownloading) {
+        const std::string& identifier = parent()->identifier();
+        _dynamicFileDownloader = std::make_unique<DynamicFileSequenceDownloader>(
+            _dataID, identifier, _infoURL, _dataURL, _nFilesToQueue
+        );
+    }
+    if (_loadingType == LoadingType::DynamicDownloading) {
+        updateDynamicDownloading(currentTime, deltaTime);
+    }
+
+    _inInterval = !_files.empty() &&
+        currentTime >= _files[0].time &&
+        currentTime < _sequenceEndTime;
+
+    if (_inInterval) {
+        const size_t nextIdx = _activeTriggerTimeIndex + 1;
+        if (
+            // true => We stepped back to a time represented by another state
+            currentTime < _files[_activeTriggerTimeIndex].time ||
+            // true => We stepped forward to a time represented by another state
+            (nextIdx < _files.size() && currentTime >= _files[nextIdx].time))
+        {
+            int previousIndex = _activeTriggerTimeIndex;
+            updateActiveTriggerTimeIndex(currentTime);
+            File& file = _files[_activeTriggerTimeIndex];
+            if (file.status == File::FileStatus::Downloaded) {
+                size_t layer = _hasLayerNames ? _fitsLayerName : _fitsLayer;
+                std::pair<float, float> minMax = _layerMinMaxCaps.at(layer);
+                file.texture =
+                    loadTextureFromFits(file.path, layer, minMax);
+                file.status = File::FileStatus::Loaded;
+            }
+            if (previousIndex != _activeTriggerTimeIndex) {
+                trackOldest(file);
+                loadTexture();
+            }
+        }
+        // The case when we jumped passed last file. where nextIdx is not < file.size()
+        else if (currentTime >=
+            _files[_activeTriggerTimeIndex].time && !_texture) {
+            loadTexture();
+        }
+    }
+    if (!_firstUpdate && _useColorMap && !_files.empty()) {
+        _dataMinMaxValues = _files[_activeTriggerTimeIndex].dataMinMax;
+    }
+    if (_textureIsDirty) [[unlikely]] {
+        loadTexture();
+        _textureIsDirty = false;
+    }
+}
+
+void RenderableTimeVaryingFitsSphere::render(const RenderData& data, RendererTasks& task) {
+    if (_files.empty() || (!_inInterval && !_renderForever)) {
+        return;
+    }
+    RenderableSphere::render(data, task);
+}
+
+void RenderableTimeVaryingFitsSphere::bindTexture() {
+    if (_texture) {
+        _texture->bind();
+    }
+    else {
+        unbindTexture();
+    }
```

Comment:
```suggestion
```

Why is this here?  The function is called "bindTexture", so it should not unbind something that might have been bound before